pump/server.go

```diff
@@ -216,7 +216,7 @@ func (s *Server) writeBinlog(ctx context.Context, in *binlog.WriteBinlogReq, isF
 		goto errHandle
 	}
 
-	if !isFakeBinlog {
+	if !isFakeBinlog && blog.Tp == binlog.BinlogType_Prewrite {
```

Comment:
In the original implementation, when closing, a fake binlog is written and assumed to be the last binlog. This is true, because in this line, we guaranteed that no other non-fake binlogs could be written.

With this change, this assumption no longer holds, for example, in the following case, the commit binlog may get lost:

1. Close
1. Write Fake binlog
1. A C-Binlog is written
1. All online drainers have consumed the fake binlog but not the C-Binlog
1. The Pump is allowed to go offline and the drainers would not observe this missing C-Binlog when merging.

Reply:
add a method to make sure all p-binlog having matched c-binlog PTAL